src/vcf/record.cpp

```diff
@@ -512,9 +595,20 @@ namespace ebi
                     }
                 } else if (type == "String") {
                     // ...do nothing, it is guaranteed it will be a string
-                } 
+                }
             } catch (...) {
-                raise(std::make_shared<Error>(line, field + " does not match the meta specification Type=" + type));
+                if (status) {
```

Comment:
Ok, after thinking a while about this, I suggest the next refactor. The main idea is to remove the  `status` flag and make each function receive only the information it needs. Please consider the next 2 functions:
```    
    void check_value_type(const std::string &type, const std::string &value) {
        if (type == "Integer") {
            // ...try to cast to int
            std::stoi(value);
            // ...and also check it's not a float
            if (std::fmod(std::stof(value), 1) != 0) {
                throw std::invalid_argument("Float provided instead of Integer");
            }
        } else if (type == "Float") {
            // ...try to cast to float
            try {
                std::stof(value);
            } catch (std::out_of_range) {
                // It maybe a subnormal number
                std::stold(value);
            }
        } else if (type == "Flag") {
            int numeric_value = std::stoi(value);
            if (numeric_value != 0 && numeric_value != 1) {
                throw std::invalid_argument("A flag value must be 0 or 1");
            }
            // If no flag is provided then there is nothing to check
        } else if (type == "Character") {
            // ...check the length is 1
            if (value.size() > 1) {
                throw std::invalid_argument("There can be only one character");
            }
        } else if (type == "String") {
            // ...do nothing, it is guaranteed it will be a string
        }
    }

    void Record::check_field_type(std::string const & field,
                                  std::vector<std::string> const & values,
                                  std::string const & type) const
    {
        // To check the field type...
        for (auto & value : values) {
            if (value == ".") { continue; }
            
            try {
                check_value_type(type, value);
            } catch (std::exception &typeError) {
                throw std::invalid_argument(field + " does not match the meta specification Type=" + type + ". " 
                                                    + typeError.what());
            }
        }
    }
```
As you see, the first one only knows about one value (if a field is "AC=3,4,5" this function will receive "Integer" and "3", for instance), and consequently, it raises an exception containing that information. In the second function, we are in the context of a type check for several values, so it add that information to the exception message and rethrows. finally, in some place like `check_info`, the second function will be called, and will catch an exception and will add to the message that it is a "INFO" check (that's the only place that should know it), and only then build an Error from our hierarchy, a `InfoBodyError` in this case, reusing the message from the inner exception.

A similar thing can be done for `check_predefined_tag`.